File: backend/src/main/java/com/ll/hotel/domain/member/admin/dto/response/AdminHotelResponse.java
package com.ll.hotel.domain.member.admin.dto.response;

import com.ll.hotel.domain.hotel.hotel.entity.Hotel;
import com.ll.hotel.domain.hotel.hotel.type.HotelStatus;
import com.ll.hotel.domain.hotel.option.entity.HotelOption;
import com.ll.hotel.domain.member.member.entity.Business;
import com.ll.hotel.domain.member.member.entity.Member;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;

public class AdminHotelResponse {
    public record ApprovalResult(
            String name,
            HotelStatus status
    ) {
        public static ApprovalResult from(Hotel hotel) {
            return new ApprovalResult(
                    hotel.getHotelName(),
                    hotel.getHotelStatus()
            );
        }
    }

    public record Summary(
            Long hotelId,
            String name,
            String streetAddress,
            String ownerName,
            HotelStatus status
    ) {
        public static Summary from(Hotel hotel) {
            Business business = hotel.getBusiness();
            Member owner = business.getMember();
            return new Summary(
                    hotel.getId(),
                    hotel.getHotelName(),
                    hotel.getStreetAddress(),
                    owner.getMemberName(),
                    hotel.getHotelStatus()
            );
        }
    }

    public record Detail(
            String hotelName,
            String streetAddress,
            Integer zipCode,
            Integer hotelGrade,
            LocalTime checkInTime,
            LocalTime checkOutTime,
            HotelStatus hotelStatus,
            
            String hotelEmail,
            String hotelPhoneNumber,

            Long ownerId,
            String ownerName,
            String businessRegistrationNumber,
            LocalDate startDate,

            Double averageRating,
            Long totalReviewCount,

            Set<String> hotelOptions
    ) {
        public static Detail from(Hotel hotel) {
            Business business = hotel.getBusiness();
            return new Detail(
                    hotel.getHotelName(),
                    hotel.getStreetAddress(),
                    hotel.getZipCode(),
                    hotel.getHotelGrade(),
                    hotel.getCheckInTime(),
                    hotel.getCheckOutTime(),
                    hotel.getHotelStatus(),

                    hotel.getHotelEmail(),
                    hotel.getHotelPhoneNumber(),

                    business.getId(),
                    business.getOwnerName(),
                    business.getBusinessRegistrationNumber(),
                    business.getStartDate(),

                    hotel.getAverageRating(),
                    hotel.getTotalReviewCount(),
                    hotel.getHotelOptions() != null
                            ? hotel.getHotelOptions().stream()
                            .map(HotelOption::getName)
                            .collect(Collectors.toSet())
                            : new HashSet<>()
            );
        }
    }
}

File: backend/src/main/java/com/ll/hotel/domain/member/admin/dto/response/AdminHotelResponse.kt
package com.ll.hotel.domain.member.admin.dto.response

import com.ll.hotel.domain.hotel.hotel.entity.Hotel
import com.ll.hotel.domain.hotel.hotel.type.HotelStatus
import java.time.LocalDate
import java.time.LocalTime

class AdminHotelResponse {
    data class ApprovalResult(
        val name: String,
        val status: HotelStatus
    ) {
        companion object {
            fun from(hotel: Hotel) = ApprovalResult(
                name = hotel.hotelName,
                status = hotel.hotelStatus
            )
        }
    }

    data class Summary(
        val hotelId: Long,
        val name: String,
        val streetAddress: String,
        val ownerName: String,
        val status: HotelStatus
    ) {
        companion object {
            fun from(hotel: Hotel) = Summary(
                hotelId = hotel.id,
                name = hotel.hotelName,
                streetAddress = hotel.streetAddress,
                ownerName = hotel.business.member.memberName,
                status = hotel.hotelStatus
            )
        }
    }

    data class Detail(
        val hotelName: String,
        val streetAddress: String,
        val zipCode: Int,
        val hotelGrade: Int,
        val checkInTime: LocalTime,
        val checkOutTime: LocalTime,
        val hotelStatus: HotelStatus,

        val hotelEmail: String,
        val hotelPhoneNumber: String,

        val ownerId: Long,
        val ownerName: String,
        val businessRegistrationNumber: String,
        val startDate: LocalDate,

        val averageRating: Double,
        val totalReviewCount: Long,

        val hotelOptions: Set<String>?
    ) {
        companion object {
            fun from(hotel: Hotel) = Detail(
                hotelName = hotel.hotelName,
                streetAddress = hotel.streetAddress,
                zipCode = hotel.zipCode,
                hotelGrade = hotel.hotelGrade,
                checkInTime = hotel.checkInTime,
                checkOutTime = hotel.checkOutTime,
                hotelStatus = hotel.hotelStatus,

                hotelEmail = hotel.business.member.memberEmail,
                hotelPhoneNumber = hotel.business.member.memberPhoneNumber,

                ownerId = hotel.business.member.id,
                ownerName = hotel.business.member.memberName,
                businessRegistrationNumber = hotel.business.businessRegistrationNumber,
                startDate = hotel.business.startDate,

                averageRating = hotel.averageRating,
                totalReviewCount = hotel.totalReviewCount,
                hotelOptions = hotel.hotelOptions?.map { it.name }?.toSet() ?: emptySet()
            )
        }
    }
}
